core/include/drivers/firewall_device.h

```diff
@@ -0,0 +1,224 @@
+/**
+ * firewall_dt_get_by_index - Get the firewall configuration associated to a
+ * given index for a device node.
+ *
+ * @fdt: FDT to work on
+ * @node: Device node to read from
+ * @index: Index of the entry in the property
+ * @out_fw: Firewall reference
+ *
+ * Returns TEE_SUCCESS on success, TEE_ERROR_ITEM_NOT_FOUND if there's no match
+ * with a firewall controller or appropriate TEE_Result error code if an
+ * error occurred.
+ */
+TEE_Result firewall_dt_get_by_index(const void *fdt, int node,
+				    unsigned int index,
+				    struct firewall_query **out_fw);
+
+/**
+ * firewall_dt_get_by_name - Get the firewall configuration associated to a
+ * given name for a device node.
+ *
+ * @fdt: FDT to work on
+ * @node: Device node to read from
+ * @name: Name of the firewall configuration to search for
+ * @out_fw: Firewall reference
+ *
+ * Returns TEE_SUCCESS on success, TEE_ERROR_ITEM_NOT_FOUND if there's no match
+ * with a firewall controller or appropriate TEE_Result error code if an
+ * error occurred.
+ */
+TEE_Result firewall_dt_get_by_name(const void *fdt, int node, const char *name,
+				   struct firewall_query **out_fw);
+
+/**
+ * firewall_set_configuration - Reconfigure the firewall controller associated
+ * to the given firewall configuration with it.
+ *
+ * @fw:	Firewall query containing the configuration to set
+ *
+ * Returns TEE_SUCCESS if configuration is correctly set,
+ * TEE_ERROR_ITEM_NOT_FOUND if there's no configuration to set or appropriate
+ * TEE_Result error code if an error occurred.
+ */
+TEE_Result firewall_set_configuration(struct firewall_query *fw);
+
+/**
+ * firewall_check_access - Check if the access is authorized for the consumer
+ * and the given firewall settings according to the configuration of its
+ * firewall controller
+ *
+ * @fw:	Firewall query to check against its firewall controller
+ */
+TEE_Result firewall_check_access(struct firewall_query *fw);
+
+/**
+ * firewall_acquire_access - Check if OP-TEE can access the consumer and
+ * acquire potential resources to allow the access
+ *
+ * @fw:	Firewall query to check against its firewall controller
+ */
+TEE_Result firewall_acquire_access(struct firewall_query *fw);
+
+/**
+ * firewall_check_memory_access - Check if a consumer can access, in read or
+ * write, the given memory range against a firewall controller
+ *
+ * @fw: Firewall configuration to check against its firewall controller
+ * @paddr: Physical base address of the memory range to check
+ * @size: Size of the memory range to check
+ * @read: Check rights for a read access
+ * @write: Check rights for a write access
+ */
+TEE_Result firewall_check_memory_access(struct firewall_query *fw,
+					paddr_t paddr, size_t size, bool read,
+					bool write);
+
+/**
+ * firewall_acquire_memory_access - Check if OP-TEE can access, in read or
+ * write, the given memory range against a firewall controller and acquire
+ * potential resources to allow the access
```

Comment:
Instead of `Check if` I would rather say `Request OP-TEE access, in read and/or write mode, to memory address range`.